source/Mlos.Model.Services.Client/OptimizerServiceEncoderDecoder.cs

```diff
@@ -0,0 +1,419 @@
+// -----------------------------------------------------------------------
+// <copyright file="OptimizerServiceEncoderDecoder.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root
+// for license information.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using Mlos.Model.Services.Spaces;
+
+namespace Mlos.Model.Services.Client
+{
+    public class OptimizerServiceEncoder
+    {
+        public OptimizerService.SimpleHypergrid EncodeHypergrid(Hypergrid hypergrid)
+        {
+            OptimizerService.SimpleHypergrid instance = new OptimizerService.SimpleHypergrid();
+
+            // TODO: I have no idea if this is the right way to handle subgrids.

```

Comment:
Is this TODO comment still timely?
There are several examples on how to handle them on the Python Side (encoders/decoders should be closest to your scenario). Also, testing may reveal gaps.